include/core/interface_tools.h

```diff
@@ -286,6 +304,414 @@ namespace InterfaceTools
+      std::shared_ptr<FiniteElement<dim>> background_fe,
+      const double                        p_max_distance,
+      const double                        p_iso_level)
+      : dof_handler(*background_triangulation)
+      , fe(background_fe)
+      , max_distance(p_max_distance)
+      , iso_level(p_iso_level)
+      , pcout(std::cout,
+              (Utilities::MPI::this_mpi_process(MPI_COMM_WORLD) == 0))
+    {
+      mapping = std::make_shared<MappingFE<dim>>(*fe);
+    }
+
+    /**
+     * @brief setup_dofs
+     *
+     * Initialize the degree of freedom and the memory
+     * associated with them
+     */
+    void
+    setup_dofs(const MPI_Comm &mpi_communicator);
+
+    /**
+     * @brief set_level_set_from_background_mesh
+     *
+     * Set the level-set field from the main solver. For example, when using the
+     * current SignedDistanceSolver for the geometric redistanciation of the VOF
+     * phase fraction, the level-set field comes from the VOF solver and is
+     * described by the corresponding VOF DoFHandler.
+     *
+     * @param[in] background_dof_handler DoFHandler corresponding to the
+     * level-set field solver
+     *
+     * @param[in] background_level_set_vector level-set solution vector
+     *
+     * @param[in] mpi_communicator MPI communicator
+     */
+    void
+    set_level_set_from_background_mesh(
+      const DoFHandler<dim> &background_dof_handler,
+      const VectorType      &background_level_set_vector,
+      const MPI_Comm        &mpi_communicator);
+
+    /**
+     * @brief solve
+     *
+     * Solve for the signed distance from the given level of the level-set
+     * vector.
+     */
+    void
+    solve(const MPI_Comm &mpi_communicator);
+
+    /**
+     * @brief Getter methods to get the private attribute signed_distance
+     *
+     * @param[in] mpi_communicator MPI communicator
+     *
+     * @return vector storing the computed signed distance
+     */
+    VectorType &
+    get_signed_distance(const MPI_Comm &mpi_communicator);
+
+    /**
+     * @brief Output the interface reconstruction used for the signed distance
+     * computations
+     *
+     * @param[in] output_name name of the output file
+     *
+     * @param[in] output_path path to the output file
+     *
+     * @param[in] it iteration number
+     *
+     */
+    void
+    output_interface_reconstruction(const std::string  output_name,
+                                    const std::string  output_path,
+                                    const unsigned int it) const;
+
+    /**
+     * @brief Attach the solution vector to the DataOut provided. This function
+     * enable the auxiliary physics to output their solution via the background
+     * solver.
+     *
+     * @param[in,out] data_out DataOut reponsible for solution output
+     */
+    void
+    attach_solution_to_output(DataOut<dim> &data_out);
+
+    /// DoFHandler describing the problem
+    DoFHandler<dim> dof_handler;
```

Comment:
Is there a reason why the dof_handler is public? You use it elsewhere I guess?

Reply:
It is to use it when we will interpolate the signed distance to the main solver (VOF) dofhandler

Reply:
I can add a get_dof_handler method also :)